src/paulscode/android/mupen64plusae/input/map/TouchMap.java

```diff
@@ -265,6 +271,28 @@ public int getButtonPress( int xLocation, int yLocation )
     }
     
     /**
+     * Gets the frame for the N64 button with a given asset name
+     * 
+     * @param assetName The asset name for the button
+     * 
+     * @return The frame for the N64 button with the given asset name
+     * 
+     */
+    public Rect getButtonFrame( String assetName )
```

Comment:
Will do. The main problem here was that getButtonPress loops over the buttons to see which one intersects the touch input, but then it discards that information in favor of returning the N64 button index associated with the color written to what I assume is an off-screen image. I think I originally had it returning one of those rects, but by the time I figured out how to map the return value back to buttonMasks I had long since abandoned that code.